src/causalprog/utils/translator.py

```diff
@@ -0,0 +1,141 @@
+"""
+Helper class to keep the codebase backend-agnostic.
+
+Our frontend (or user-facing) classes each use a syntax that applies across the package
+codebase. By contrast, the various backends that we want to support will have different
+syntaxes and call signatures for the functions that we want to support. As such, we need
+a helper class that can store this "translation" information, allowing the user to
+interact with the package in a standard way but also allowing them to choose their own
+backend if desired.
+"""
+
+import inspect
+from abc import ABC, abstractmethod
+from typing import Any
+
+
+class Translator(ABC):
+    """
+    Maps syntax of a backend function to our frontend syntax.
+
+    Different backends have different syntax for drawing samples from the distributions
+    they support. In order to map these different syntaxes to our backend-agnostic
+    framework, we need a container class to map the names we have chosen for our
+    frontend methods to those used by their corresponding backend method.
+
+    A ``Translator`` allows us to identify whether a user-provided backend object is
+    compatible with one of our frontend wrapper classes (and thus, call signatures). It
+    also allows users to write their own translators for any custom backends that we do
+    not explicitly support.
+
+    The use case for a ``Translator`` is as follows. Suppose that we have a frontend
+    class ``C`` that needs to provide a method ``do_something``. ``C`` stores a
+    reference to a backend object ``obj`` that can provide the functionality of
+    ``do_something`` via one of its methods, ``obj.backend_method``. However, there is
+    no guarantee that the signature of ``do_something`` maps identically to that of
+    ``obj.backend_method``. A ``Translator`` allows us to encode a mapping of
+    ``obj.backend_method``s arguments to those of ``do_something``.
+    """
+
+    backend_method: str
+    corresponding_backend_arg: dict[str, str]
+
+    @property
+    @abstractmethod
+    def _frontend_method(self) -> str:
+        """Name of the frontend method that the backend is to be translated into."""
+        return ""
```

Comment:
Abstract methods and properties do not need to return anything. We could simplify this to:

```suggestion
        """Name of the frontend method that the backend is to be translated into."""
```